src/dbnode/storage/index.go

```diff
@@ -653,15 +654,19 @@ func (i *nsIndex) writeBatches(
 			if !ts.After(pastLimit) {
 				// NB(bodu): We only mark entries as too far in the past if
 				// cold writes are not enabled.
-				var err error
 				if !i.coldWritesEnabled {
-					err = m3dberrors.ErrTooPast
+					batch.MarkUnmarkedEntryError(m3dberrors.ErrTooPast, idx)
+					return
+				}
+				// We allow index writes that are between mutable block start and
+				// start of buffer past to go through.
+				if !ts.After(warmBlockStart) {
```

Comment:
Can we use `if ts.Before(warmBlockStart)` since we want to allow indexing to be inclusive of the start time of the block (exact match on block start timestamp).

Reply:
Also, might be worth a test here to prove that this is possible.

Reply:
We can change `if !ts.After(pastLimit) {` on line 654 to `if ts.Before(pastLimit) {` as well to match the same logic.

Reply:
I changed the test index write gen code to start generating writes right at the warm index block start.
https://github.com/m3db/m3/pull/2329/files#diff-bc6de8fe8564769dc8e8d73f46cc104fR132

Reply:
The first write tests that writes exactly at the boundary are inclusive.

Reply:
Nice, sg